app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -174,13 +180,22 @@ def audit_failure(err)
           Audit::Event::Authn::InjectClientCert.new(
             authenticator_name: AUTHENTICATOR_NAME,
             service:            webservice,
-            role_id:            host.id,
+            role_id:            sanitized_role_id,
             client_ip:          @client_ip,
             success:            false,
             error_message:      err.message
           )
         )
       end
+
+      # TODO: this logic is taken from app/models/audit/event/authn.rb.
+      # We should use that logic here.
+      # Masking role if it doesn't exist to avoid audit pollution
+      # Checking @success as well to save DB call on success
```

Comment:
you actually don't check @success here. Please fix comment. I would write that we're doing this as `host` doesn't exist so it doesn't have an id.